src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/RuntimeExports.cs

```diff
@@ -166,25 +166,21 @@ public static unsafe void RhUnboxAny(object? o, ref byte data, MethodTable* pUnb
         {
             if (pUnboxToEEType->IsValueType)
             {
-                bool isValid = false;
+                bool isValid;
 
                 if (pUnboxToEEType->IsNullable)
                 {
                     isValid = (o == null) || o.GetMethodTable() == pUnboxToEEType->NullableType;
                 }
                 else
                 {
-                    isValid = (o != null) && UnboxAnyTypeCompare(o.GetMethodTable(), pUnboxToEEType);
+                    // This will throw NullReferenceException if obj is null.
+                    isValid = UnboxAnyTypeCompare(o.GetMethodTable(), pUnboxToEEType);
                 }
 
                 if (!isValid)
                 {
-                    // Throw the invalid cast exception defined by the classlib, using the input unbox MethodTable*
-                    // to find the correct classlib.
-
-                    ExceptionIDs exID = o == null ? ExceptionIDs.NullReference : ExceptionIDs.InvalidCast;
```

Comment:
I think a several of these ExceptionIDs are now unused (e.g. `ExceptionIDs.InvalidCast` is only used for switch statements that handle impossible situations).

Reply:
Removed.